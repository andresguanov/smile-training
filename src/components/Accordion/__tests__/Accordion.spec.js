import { shallowMount } from "@vue/test-utils"
import smAccordion from "@components/Accordion/Accordion.vue"
let wrapper
const sections = [
  {
    title: "Title number 1 for section",
    content: "Lorem ipsum dolor sit amet, consectetur adipisicing elit"
  },
  {
    title: "Maybe something different",
    content:
      "sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam"
  },
  {
    title: "Last one for the road",
    content: "Hit the road jack, and dont you look back, no more"
  }
]
const $id = id => id

describe("smAccordion", () => {
  //#region Basic render
  it("renders default accordion", () => {
    wrapper = shallowMount(smAccordion, {
      propsData: {
        sections
      },
      mocks: {
        $id
      }
    })
    expect(wrapper.classes("bg-brand1")).toBe(true)
    expect(wrapper.findAll("li.accordion-section")).toHaveLength(sections.length)
    expect(wrapper.html()).toMatchSnapshot()
  })
  //#endregion

  //#region Props
  it("sets defaultOpen", () => {
    wrapper = shallowMount(smAccordion, {
      propsData: {
        sections,
        defaultOpen: sections[2].title
      },
      mocks: {
        $id
      }
    })
    expect(wrapper.vm.open).toBe(sections[2].title)
  })

  it("sets navClass", () => {
    const navClass = "custom-class"
    wrapper = shallowMount(smAccordion, {
      propsData: {
        sections,
        navClass
      },
      mocks: {
        $id
      }
    })
    expect(wrapper.classes(navClass)).toBe(true)
  })
  //#endregion

  //#region Methods
  it("sets defaultOpen through", async () => {
    wrapper = shallowMount(smAccordion, {
      propsData: {
        sections,
        defaultOpen: sections[2].title
      },
      mocks: {
        $id
      }
    })
    expect(wrapper.vm.open).toBe(sections[2].title)
    await wrapper.findAll("label").at(1).trigger("click")
    expect(wrapper.vm.open).toBe(sections[1].title)
  })
  //#endregion

  //#region Events
  //#endregion

  //#region Slots
  it("receives title and content on slots", async () => {
    wrapper = shallowMount(smAccordion, {
      propsData: {
        sections
      },
      mocks: {
        $id
      },
      scopedSlots: {
        title: "<p>{{props.title}}</p>",
        default: "<p>{{props.content}}</p>"
      }
    })
    const titles = wrapper.findAll("label p").wrappers.map(l => l.text())
    const actualTitles = sections.map(s => s.title)
    expect(titles).toMatchObject(actualTitles)
    const contents = wrapper.findAll("ul.accordion-submenu p").wrappers.map(l => l.text())
    const actualContents = sections.map(s => s.content)
    expect(contents).toMatchObject(actualContents)
  })
  //#endregion
})
