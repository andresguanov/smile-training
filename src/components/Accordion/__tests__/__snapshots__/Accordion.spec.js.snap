// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smAccordion renders default accordion 1`] = `
<form class="bg-brand1">
  <ul class="accordion">
    <li class="accordion-section has-submenu"><input id="smAccordion-0" name="main-nav" type="radio" value="Title number 1 for section"> <label for="smAccordion-0"><span style="margin-left: 0px;">Title number 1 for section</span></label>
      <ul class="accordion-submenu">
        <div style="padding: 16px 24px;">
          Lorem ipsum dolor sit amet, consectetur adipisicing elit
        </div>
      </ul>
    </li>
    <li class="accordion-section has-submenu"><input id="smAccordion-1" name="main-nav" type="radio" value="Maybe something different"> <label for="smAccordion-1"><span style="margin-left: 0px;">Maybe something different</span></label>
      <ul class="accordion-submenu">
        <div style="padding: 16px 24px;">
          sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam
        </div>
      </ul>
    </li>
    <li class="accordion-section has-submenu"><input id="smAccordion-2" name="main-nav" type="radio" value="Last one for the road"> <label for="smAccordion-2"><span style="margin-left: 0px;">Last one for the road</span></label>
      <ul class="accordion-submenu">
        <div style="padding: 16px 24px;">
          Hit the road jack, and dont you look back, no more
        </div>
      </ul>
    </li>
  </ul>
</form>
`;
